File: src/ui/src/models/api/ApiKey/ApiKeySummaryModel.ts
/**
 * Summary model for API key
 */
export interface ApiKeySummaryModel {
  /**
   * Unique identifier
   */
  id: string;

  /**
   * When the API key was created
   */
  createdAt: string;

  /**
   * Name of the API key
   */
  name: string;

  /**
   * Optional description
   */
  description?: string;

  /**
   * Whether the key is enabled
   */
  isEnabled: boolean;

  /**
   * Masked API key value (for display only)
   */
  apiKey: string;
}

File: src/backend/DonkeyWork.Chat.Api/Models/ApiKey/ApiKeySummaryModel.cs
// ------------------------------------------------------
// <copyright file="ApiKeySummaryModel.cs" company="DonkeyWork.Dev">
// Provided as is by (c) DonkeyWork.Dev. No warranty or support is given.
// </copyright>
// ------------------------------------------------------

namespace DonkeyWork.Chat.Api.Models.ApiKey;

/// <summary>
/// A summary of an API key.
/// </summary>
public record ApiKeySummaryModel
{
    /// <summary>
    /// Gets the id of the api key.
    /// </summary>
    public Guid Id { get; init; }

    /// <summary>
    /// Gets the created date of the API key.
    /// </summary>
    public DateTimeOffset CreatedAt { get; init; }

    /// <summary>
    /// Gets the name of the API key.
    /// </summary>
    public string Name { get; init; } = string.Empty;

    /// <summary>
    /// Gets the description of the API key.
    /// </summary>
    public string? Description { get; init; }

    /// <summary>
    /// Gets a value indicating whether the API key is enabled.
    /// </summary>
    public bool IsEnabled { get; init; }

    /// <summary>
    /// Gets a hashed partial value of the api key.
    /// </summary>
    public string ApiKey { get; init; } = string.Empty;
}